OmniKit/PumpManager/OmnipodPumpManager.swift

```diff
@@ -907,74 +907,25 @@ extension OmnipodPumpManager {
-
-        result += String(format: LocalizedString("Delivery Status: %1$@\n", comment: "The format string for Delivery Status: (1: delivery status string)"), String(describing: status.deliveryStatus))
-
-        if let lastInsulinMeasurements = self.state.podState?.lastInsulinMeasurements {
-            delivered = lastInsulinMeasurements.delivered
-        } else {
-            delivered = status.insulin
-        }
-        result += String(format: LocalizedString("Total Insulin Delivered: %1$@ U\n", comment: "The format string for Total Insulin Delivered: (1: total insulin delivered string)"), delivered.twoDecimals)
-
-        result += String(format: LocalizedString("Reservoir Level: %1$@ U\n", comment: "The format string for Reservoir Level: (1: reservoir level string)"), status.reservoirLevel?.twoDecimals ?? "50+")
-
-        result += String(format: LocalizedString("Last Bolus Not Delivered: %1$@ U\n", comment: "The format string for Last Bolus Not Delivered: (1: bolus not delivered string)"), status.insulinNotDelivered.twoDecimals)
-
-        if let podState = self.state.podState,
-            podState.activeAlerts.startIndex != podState.activeAlerts.endIndex
-        {
-            // generate a more helpful string with the actual alert names
-            str = String(describing: podState.activeAlerts)
-        } else {
-            str = String(describing: status.alerts)
-        }
-        result += String(format: LocalizedString("Alerts: %1$@\n", comment: "The format string for Alerts: (1: the alerts string)"), str)
-
-        return result
-    }
-    
-    public func readPodStatus(completion: @escaping (String) -> Void) {
+    public func readPodStatus(completion: @escaping (Result<DetailedStatus, Error>) -> Void) {
         // use hasSetupPod to be able to read pod info from a faulted Pod
         guard self.hasSetupPod else {
-            completion(PodCommsError.noPodPaired.errorDescription!)
+            completion(.failure(PodCommsError.noPodPaired))
             return
         }
 
         let rileyLinkSelector = self.rileyLinkDeviceProvider.firstConnectedDevice
         podComms.runSession(withName: "Read pod status", using: rileyLinkSelector) { (result) in
-            
-            let reportError = { (errorStr: String) in
-                self.log.error("Failed to read pod status: %{public}@", errorStr)
-                completion(errorStr)
-            }
-            
             do {
                 switch result {
                 case .success(let session):
-                    let status = try session.getStatus()
-                    session.dosesForStorage({ (doses) -> Bool in
-                        self.store(doses: doses, in: session)
-                    })
-                    completion(self.podStatusString(status: status))
+                    let faultStatus = try session.getDetailedStatus()
+                    completion(.success(faultStatus))
```

Comment:
I am not comfortable with leaving out the functionality of the podState.updateFromStatusResponse() in session.getStatus() and the session.dosesForStorage() out for the enhanced Read Pod Status as uses like to use this command to update all the stats.

Reply:
I added a refresh button to the pod settings page.

Reply:
Also, see https://github.com/LoopKit/Loop/pull/1381

Reply:
The refresh button is a great addition to have -- especially near the top of the Pod Settings View. But it seems silly/unintuitive/confusing/wrong to have a manual Red Pod Status diagnostic command which doesn't actually update Loop's pod state which is displayed in Pod Settings (Active Time, Alarms, Reservoir, Insulin Delivered, etc.) immediately upon return from the Read Pod Status view as it has always done before. It be hard to diagnose anything when it's reporting one thing and Pod Settings is displaying something else. I already have implemented a new version of updateFromStatusResponse() which works with either a StatusResponse and a DetailedStatus and cleans up some other related code in PR #624. I'd like to do another commit on this PR to include this code and have the Read Pod Status command use it.